cmd/nitro/nitro.go

```diff
@@ -351,19 +354,33 @@ func mainImpl() int {
 		if err != nil {
 			log.Crit("error getting rollup addresses", "err", err)
 		}
+		arbSys, _ := precompilesgen.NewArbSys(types.ArbSysAddress, l1Client)
+		l1Reader, err = headerreader.New(ctx, l1Client, func() *headerreader.Config { return &liveNodeConfig.Get().Node.ParentChainReader }, arbSys)
+		if err != nil {
+			log.Crit("failed to get L1 headerreader", "err", err)
+		}
+		if !l1Reader.IsParentChainArbitrum() && !nodeConfig.Node.Dangerous.DisableBlobReader {
```

Comment:
instead of having a DisableBlobReader and test for IsParentChainArbitrum.. let's do a RequireBlobReader option, and have it be set from arbitrum_chain_info. Default should probably be false, but both arb1 and arbnova will set it to true from arbitrum_chain_info.

Reply:
Why put it in the chain info? It seems to me like it's better to auto detect this than let chain providers misconfigure it.

Reply:
It is essentially a parameter describing the chain, and belongs there next to feed_url and has-genesis-state. It's not harder for orbit chains to config then those.
Testnode is a great example for a chain that's not on arbitrum and shouldn't require blob. Orbit could be on non-arbitrum L2 that either supports Blobs or doesn't.

Reply:
There's definitely more configurability we need in terms of non-Arbitrum L2s, but I think we should approach that as a whole and probably add a new chain info fields to try to capture more of these variables at once. New chain info fields need to be communicated to rollup as a service providers, so I'd prefer to figure out a comprehensive strategy there and then message it. As-is, I'd worry most RaaS providers wouldn't configure this field properly, and right now there's no chain we support that would need an exception to this rule apart from the test node. I don't think other L2s would work properly right now, e.g. L1 data charging, and who knows what other edge cases they have in regards to block numbers and headers and such. Maybe Polygon would work but not support blobs? I'm not sure. Regardless, I think we should just have the flag for now, and then we can circle back and add the chain info option if we need it. I think it's good to have the flag even if we had the chain info option for local testing and debugging, so I think it's a good starting point at least.